sdk/database/dbplugin/plugin.go

```diff
@@ -44,6 +44,19 @@ type Database interface {
 	// the API.
 	RotateRootCredentials(ctx context.Context, statements []string) (config map[string]interface{}, err error)
 
+	// GenerateCredentials returns a generated password for the plugin. This is
+	// used in combination with SetCredentials to set a specific password for a
+	// database user and preserve the password in WAL entries.
+	GenerateCredentials(ctx context.Context) (string, error)
+
+	// SetCredentials uses provided information to create or set the credentials
+	// for a database user. Unlike CreateUser, this method requires both a
+	// username and a password given instead of generating them. This is used for
+	// creating and setting the password of static accounts, as well as rolling
+	// back passwords in the database in the event an updated database fails to
+	// save in Vault's storage.
+	SetCredentials(ctx context.Context, statements Statements, staticConfig StaticUserConfig) (username string, password string, err error)
+
```

Comment:
Have you tested that an old plugin that doesn't satisfy these interfaces still works? I think it should, but would be good to make sure.

Reply:
I have tested it, you get an error saying the plugin doesn't support static accounts

Reply:
Granted, I did this manually, so I have no automated/scripted way to prove this to you at this time